src/rest/APIRequest.js

```diff
@@ -33,7 +33,7 @@ class APIRequest {
         ? this.client.options.http.api
         : `${this.client.options.http.api}/v${this.client.options.http.version}`;
     const url = API + this.path;
-    let headers = {};
+    let headers = this.client.options.http.headers ?? {};
```

Comment:
Actually, now that I think of it, doesn't this need to be cloned? Otherwise, after the first run, it'll be mutated and the options will leak into next requests, e.g. the reason header.

Reply:
```suggestion
    let headers = { ...this.client.options.http.headers ?? {} };
```